homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -0,0 +1,120 @@
+"""DataUpdateCoordinator for the Yale integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from yalesmartalarmclient.client import AuthenticationError, YaleSmartAlarmClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    STATE_LOCKED,
+    STATE_UNAVAILABLE,
+    STATE_UNLOCKED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, LOGGER
+
+
+class YaleDataUpdateCoordinator(DataUpdateCoordinator):
+    """A Yale Data Update Coordinator."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize the Verisure hub."""
+        self.entry = entry
+        self._hass = hass
+        self.yale = None
+
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict:
+        """Fetch data from Yale."""
+
+        updates = await self._hass.async_add_executor_job(self.get_updates)
+
+        locks = []
+        door_windows = []
+
+        for lock in updates["cycle"]["data"]["device_status"]:
+            if lock["type"] == "device_type.door_lock":
+                state = lock["status1"]
+                lock_status_str = lock["minigw_lock_status"]
+                if lock_status_str != "":
+                    lock_status = int(lock_status_str, 16)
+                    closed = (lock_status & 16) == 16
+                    locked = (lock_status & 1) == 1
+                    if closed is True and locked is True:
+                        state = STATE_LOCKED
```

Comment:
Please don't use Home Assistant constants for the device api states. We can define local constants if needed.